src/System.Text.Primitives/System/Text/Encoding/Utf8/Utf8Encoder.cs

```diff
@@ -56,6 +56,25 @@ public static bool TryDecode(ReadOnlySpan<byte> utf8, Span<char> utf16, out int
             bytesConsumed = 0;
             charactersWritten = 0;
 
+            if (Vector.IsHardwareAccelerated)
+            {
+                var utf8Vectors = utf8.NonPortableCast<byte, Vector<byte>>();
+                
+                int asciiLoops = Math.Min(utf8.Length, utf16.Length) / Vector<byte>.Count, asciiLoop = 0, charOffset = 0;
+                while (asciiLoops-- != 0 && (utf8Vectors[asciiLoop] & _highBit) == Vector<byte>.Zero)
```

Comment:
@shiftylogic  I'll play with it tomorrow (1am here), but initial results make the indexer approach look faster: https://docs.google.com/spreadsheets/d/1kirigkHeafFH_bvqIZBPoyi9lep5F4TYTFmNoLtLOHc/edit#gid=0